backend/infrahub/graphql/mutations/proposed_change.py

```diff
@@ -150,6 +150,8 @@ async def mutate_update(
                 raise ValidationError(str(exc)) from exc
 
         if updated_state == ProposedChangeState.MERGED:
+            if obj.is_draft.value:
+                raise ValidationError("A draft proposed change is not allowed to be merged")
```

Comment:
I wonder if the checks performed allowing a PR to be merge inside the mutation should be a some point moved to the `ProposedChangeChecker.verify_proposed_change_is_mergeable` method instead; and call that method in the mutation.

If so, it's probably for another PR.